src/Symfony/Component/Lock/Lock.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock;
+
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\NullLogger;
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockReleasingException;
+use Symfony\Component\Lock\Store\BlockingStoreInterface;
+use Symfony\Component\Lock\Store\ExpirableStoreInterface;
+
+/**
+ * Lock is the default implementation of the LockInterface.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+final class Lock implements LockInterface
+{
+    use LoggerAwareTrait;
+
+    private $store;
+    private $key;
+
+    public function __construct(Key $key, StoreInterface $store)
+    {
+        $this->store = $store;
+        $this->key = $key;
+
+        $this->logger = new NullLogger();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function acquire($blocking = false, $ttl = 300.0)
+    {
+        try {
+            if (!$blocking) {
+                $this->store->save($this->key);
+            } elseif ($this->store instanceof BlockingStoreInterface) {
+                $this->store->waitAndSave($this->key);
+            } else {
+                throw new InvalidArgumentException(sprintf('Unable to acquire a blocking lock on a instance of "%s".', get_class($this->store)));
+            }
+
+            $this->logger->info('Lock successfully acquired for "{resource}"', array('resource' => $this->key->getResource()));
+            $this->refresh($ttl);
+        } catch (LockConflictedException $e) {
+            $this->logger->info('Fail to lock the "{resource}". Someone else already acquired the lock', array('resource' => $this->key->getResource()));
+            throw $e;
+        } catch (\Exception $e) {
+            $this->logger->info('Fail to lock the "{resource}"', array('resource' => $this->key->getResource(), 'exception' => $e));
+            throw new LockAcquiringException('', 0, $e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function refresh($ttl = 300.0)
+    {
+        if (!$this->store instanceof ExpirableStoreInterface) {
+            return;
+        }
+
+        try {
+            $this->store->expire($this->key, $ttl);
+            $this->logger->info('Expiration defined for lock "{resource}" for "{ttl}" seconds', array('resource' => $this->key->getResource(), 'ttl' => $ttl));
+        } catch (LockConflictedException $e) {
+            $this->logger->info('Fail to define a expiration for the lock "{resource}". Someone else acquired the lock', array('resource' => $this->key->getResource()));
```

Comment:
Sorry, I don't see how to merge this two sentences without removing a part (which are both important IMO) :(

`Failed to define an expiration for the "{resource}" lock because someone else acquired it`?